tests/run_tests.py

```diff
@@ -112,7 +112,7 @@ def test():
                         dest='cpp',
                         default=True,
```

Comment:
agh I see what went wrong here: because `--cpp` defaults to true, this PR actually never runs the python tests since it returns immediately after cpp tests finish... (test log in CI can confirm this). This need to be changed to `False` and  cpp tests must be run explicitly in each script. 

Sorry I think it was my bad - we sometimes forgot to run C++ tests in some test scripts so I thought let's just run them by default since they were small....